MultiLabel/SGD/src/main/java/org/tribuo/multilabel/sgd/linear/LinearSGDTrainer.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2015-2020, Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.tribuo.multilabel.sgd.linear;
+
+import com.oracle.labs.mlrg.olcut.config.Config;
+import org.tribuo.ImmutableFeatureMap;
+import org.tribuo.ImmutableOutputInfo;
+import org.tribuo.common.sgd.AbstractLinearSGDTrainer;
+import org.tribuo.common.sgd.SGDObjective;
+import org.tribuo.math.LinearParameters;
+import org.tribuo.math.StochasticGradientOptimiser;
+import org.tribuo.math.la.SGDVector;
+import org.tribuo.math.la.SparseVector;
+import org.tribuo.multilabel.MultiLabel;
+import org.tribuo.multilabel.sgd.MultiLabelObjective;
+import org.tribuo.multilabel.sgd.objectives.BinaryCrossEntropy;
+import org.tribuo.provenance.ModelProvenance;
+
+import java.util.logging.Logger;
+
+/**
+ * A trainer for a multi-label linear model which uses SGD.
+ * <p>
+ * See:
+ * <pre>
+ * Bottou L.
+ * "Large-Scale Machine Learning with Stochastic Gradient Descent"
+ * Proceedings of COMPSTAT, 2010.
+ * </pre>
+ */
+public class LinearSGDTrainer extends AbstractLinearSGDTrainer<MultiLabel,SGDVector> {
+    private static final Logger logger = Logger.getLogger(LinearSGDTrainer.class.getName());
+
+    @Config(description="The classification objective function to use.")
+    private MultiLabelObjective objective = new BinaryCrossEntropy();
+
+    /**
+     * Constructs an SGD trainer for a linear model.
+     * @param objective The objective function to optimise.
+     * @param optimiser The gradient optimiser to use.
+     * @param epochs The number of epochs (complete passes through the training data).
+     * @param loggingInterval Log the loss after this many iterations. If -1 don't log anything.
+     * @param minibatchSize The size of any minibatches.
+     * @param seed A seed for the random number generator, used to shuffle the examples before each epoch.
+     */
+    public LinearSGDTrainer(MultiLabelObjective objective, StochasticGradientOptimiser optimiser, int epochs, int loggingInterval, int minibatchSize, long seed) {
+        super(optimiser,epochs,loggingInterval,minibatchSize,seed);
+        this.objective = objective;
+    }
+
+    /**
+     * Sets the minibatch size to 1.
```

Comment:
This comment actually doesn't describe what the method is. I saw this in a couple of the other versions of the Trainer constructors too.